binderhub/static/js/index.js

```diff
@@ -101,7 +101,7 @@ function updateRepoText() {
   }
   $("#repository").attr('placeholder', text);
   $("label[for=repository]").text(text);
-  $("#ref").attr('placeholder', tag_text);
+  $("#ref").attr('placeholder', 'main');
```

Comment:
I think we may want this to be conditional on the provider. `main` is the default on github, but `master` is still the default (and not changeable, I believe) on gists. GitLab also still defaults to `master`, as does git itself.

So I think at the very least, we want `main` for github and `master` for gist. Then the question is: do we 'push' folks elsewhere to use 'main' and use that for gitlab and/or git? I'm okay either way, and would lean slightly toward the push choice given the reason for the change, and only use 'master' by default for gist where it is hardcoded and cannot be chosen by users.

Reply:
Something like bed2e0a?

Reply:
Not quite, since `tag_text` isn't "master", it's the description (a duplicate of the label). I think we need a separate variable for `placeholder` that's `main` for github, `master` for gist, and either main or master the rest (`master` would preserve existing behavior, `main` would apply the new github convention to all providers by default).

For example:

```javascript
var placeholder = "main";
if (provider == "gist") {
    placeholder = "master";
}
$("#ref").attr('placeholder', placeholder);
```

Reply:
I went for the preserving existing behaviour model as I'm unsure where other communities are on this c1b7dc6

Reply:
I think it'll be "easy" now to switch to the push choice you suggested, but maybe we prepare a blog post to release when we do it? I don't like the idea of breaking behaviour without some kind of "announcement" informing the users.

Reply:
I opened this issue to discuss this further https://github.com/jupyterhub/binderhub/issues/1175